### PR TITLE
feat: add workflow variable type "PA:NOT_EMPTY"

### PR DESCRIPTION
**Pull request by tao-xinxiu**

### CODE REVIEW

File: scheduler/scheduler-api/src/test/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/validator/NotEmptyValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.exceptions.ValidationException;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 19/08/19
+ */
+public class NotEmptyValidatorTest {
+
+    @Test
```
Comment (mboussaa) [inline scheduler/scheduler-api/src/test/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/validator/NotEmptyValidatorTest.java]: u might add tests for " abc " and "TAB" but not mandatory as isblank handle that well ;)

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/factory/NotEmptyParserValidator.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+/**
+ * @author ActiveEon Team
+ * @since 14/08/19
+ */
+public class NotEmptyParserValidator extends BaseParserValidator<String> {
+
+    public static final String NOT_EMPTY_TYPE = "NOT_EMPTY";
+
+    protected static final String NOT_EMPTY_TYPE_REGEXP = "[Nn][Oo][Tt]_[Ee][Mm][Pp][Tt][Yy]";
```
Comment (MaelAudren) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/factory/NotEmptyParserValidator.java]: I would be really good to replace static values by an enum among all ParserValidator.

File: scheduler/scheduler-api/src/test/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/validator/NotEmptyValidatorTest.java (hunk 1 above)
Comment (tao-xinxiu) [reply scheduler/scheduler-api/src/test/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/validator/NotEmptyValidatorTest.java]: Done.

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/factory/NotEmptyParserValidator.java (hunk 2 above)
Comment (tao-xinxiu) [reply scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/factory/NotEmptyParserValidator.java]: I'll do it in the next PR.

**Pull request merged**